phoenix-core/src/main/java/org/apache/phoenix/schema/transform/Transform.java

```diff
@@ -0,0 +1,220 @@
+                PhoenixDatabaseMetaData.TRANSFORM_STATUS + ", " +
+                PhoenixDatabaseMetaData.TRANSFORM_JOB_ID + ", " +
+                PhoenixDatabaseMetaData.TRANSFORM_RETRY_COUNT + ", " +
+                PhoenixDatabaseMetaData.TRANSFORM_START_TS + ", " +
+                PhoenixDatabaseMetaData.TRANSFORM_END_TS + ", " +
+                PhoenixDatabaseMetaData.OLD_METADATA + " , " +
+                PhoenixDatabaseMetaData.NEW_METADATA + " , " +
+                PhoenixDatabaseMetaData.TRANSFORM_FUNCTION +
+                " FROM " + PhoenixDatabaseMetaData.SYSTEM_TRANSFORM_NAME + " WHERE  " +
+                (Strings.isNullOrEmpty(tenantId) ? "" : (PhoenixDatabaseMetaData.TENANT_ID + " ='" + tenantId + "' AND ")) +
+                (Strings.isNullOrEmpty(schema) ? "" : (PhoenixDatabaseMetaData.TABLE_SCHEM + " ='" + schema + "' AND ")) +
+                PhoenixDatabaseMetaData.LOGICAL_TABLE_NAME + " ='" + logicalTableName + "'" +
+                (Strings.isNullOrEmpty(logicalParentName) ? "": (" AND " + PhoenixDatabaseMetaData.LOGICAL_PARENT_NAME + "='" + logicalParentName + "'" ))
+        ).executeQuery()) {
+            if (resultSet.next()) {
+                return SystemTransformRecord.SystemTransformBuilder.build(resultSet);
+            }
+            return null;
+        }
+    }
+
+    public static boolean checkIsTransformNeeded(MetaDataClient.MetaProperties metaProperties, String schemaName,
+                                                 PTable table, String logicalTableName, String parentTableName,
+                                                 String tenantId, PhoenixConnection connection) throws SQLException {
+        boolean isTransformNeeded = isTransformNeeded(metaProperties, table);
+        if (isTransformNeeded) {
+            SystemTransformRecord existingTransform = Transform.getTransformRecord(schemaName, logicalTableName, parentTableName, tenantId,connection);
+            if (existingTransform != null && existingTransform.isActive()) {
+                throw new SQLExceptionInfo.Builder(
+                        SQLExceptionCode.CANNOT_TRANSFORM_ALREADY_TRANSFORMING_TABLE)
+                        .setMessage(" Only one transform at a time is allowed ")
+                        .setSchemaName(schemaName).setTableName(logicalTableName).build().buildException();
+            }
+        }
+        return isTransformNeeded;
+    }
+
+    private static boolean isTransformNeeded(MetaDataClient.MetaProperties metaProperties, PTable table){
+        if (metaProperties.getImmutableStorageSchemeProp()!=null
+                && metaProperties.getImmutableStorageSchemeProp() != table.getImmutableStorageScheme()) {
+            // Transform is needed
+            return true;
+        }
+        if (metaProperties.getColumnEncodedBytesProp()!=null
+                && metaProperties.getColumnEncodedBytesProp() != table.getEncodingScheme()) {
+            return true;
+        }
+        return false;
```

Comment:
Could be simplified to: `return metaProperties.getColumnEncodedBytesProp() != null && metaProperties.getColumnEncodedBytesProp() != table.getEncodingScheme()`